python/ray/data/_internal/dataset_logger.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import os
+from typing import Optional
+
+import ray
+from ray._private.ray_constants import LOGGER_FORMAT
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+class DatasetLogger:
+    """Logger for Ray Datasets which, in addition to logging to stdout,
+    also writes to a separate log file at `DatasetLogger.DEFAULT_DATASET_LOG_PATH`.
+    """
+
+    DEFAULT_DATASET_LOG_PATH = "logs/datasets.log"
```

Comment:
to avoid confusion for users regarding dataset/dataset pipeline, we can name it as `logs/ray-data.log`